### PR TITLE
Add multiple projects per issues function

### LINKED ISSUE #12974: Allow multiple projects per issue
It would be really nice to be able to link an issue to multiple kanban boards.

### PR DESCRIPTION
**Pull request by tyroneyeh**
Closes #12974

Test:
https://github.com/go-gitea/gitea/assets/1255041/ae4d75f9-b853-4c81-ab0b-c10e0a904d5c

Last tested:
MultipleProjectTested

### CODE REVIEW

File: models/issues/issue_list.go [hunk 1]
```diff
@@ -230,11 +230,11 @@ func (issues IssueList) loadMilestones(ctx context.Context) error {
...
 
 func (issues IssueList) getProjectIDs() []int64 {
-	ids := make(container.Set[int64], len(issues))
+	var ids []int64
 	for _, issue := range issues {
-		ids.Add(issue.ProjectID())
+		ids = append(ids, issue.ProjectIDs()...)
 	}
-	return ids.Values()
+	return ids
```
Comment (wxiaoguang) [inline models/issues/issue_list.go]: Why it should use duplicate IDs in the list?

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 2]
```diff
@@ -173,9 +173,19 @@
...
 					{{range .OpenProjects}}
-						<a class="item muted sidebar-item-link" data-id="{{.ID}}" data-href="{{.Link}}">
+						{{$ProjectID := .ID}}
+						{{$checked := false}}
+						{{range $.Issue.Projects}}
+							{{if eq .ID $ProjectID}}
+								{{$checked = true}}
+							{{end}}
+						{{end}}
+					<a class="item muted sidebar-item-link{{if $checked}} checked{{end}}" href="#" data-id="{{.ID}}" data-href="{{$.RepoLink}}/projects/{{.ID}}">
```
Comment (wxiaoguang) [inline templates/repo/issue/view_content/sidebar.tmpl]: Does this link work for org-level projects?

File: models/issues/issue_list.go (hunk 1 above)
Comment (tyroneyeh) [reply models/issues/issue_list.go]: Modified

File: templates/repo/issue/view_content/sidebar.tmpl (hunk 2 above)
Comment (tyroneyeh) [reply templates/repo/issue/view_content/sidebar.tmpl]: Modified

File: models/issues/issue_project.go [hunk 3]
```diff
@@ -14,31 +14,33 @@ import (
...
 
-func (issue *Issue) projectID(ctx context.Context) int64 {
-	var ip project_model.ProjectIssue
-	has, err := db.GetEngine(ctx).Where("issue_id=?", issue.ID).Get(&ip)
-	if err != nil || !has {
-		return 0
+func (issue *Issue) projectIDs(ctx context.Context) []int64 {
+	var ip []project_model.ProjectIssue
+	var ips []int64
+	err := db.GetEngine(ctx).Select("project_id").Where("issue_id=?", issue.ID).Find(&ip)
```
Comment (lunny) [inline models/issues/issue_project.go]:
```suggestion
	err := db.GetEngine(ctx).Table("project_issue").Select("project_id").Where("issue_id=?", issue.ID).Find(&ips)
```

Comment (tyroneyeh) [reply models/issues/issue_project.go]: Modified,

Thanks

**Comment by tyroneyeh** [comment]
Any question and any update?

**Comment by tyroneyeh** [comment]
Conflict has arisen, do not want to resolve, close

**Pull request closed without merging**